### PR TITLE
♻️(backends) add logger property to data/lrs backends

### PR DESCRIPTION
**Pull request by SergioSim**
## Purpose

Currently, backends use a module-level logger instance for logging to allow filtering logs by backend.
However, when a backend inherits another backend, logs produced by the parent backend refer to the parent module (and not the current backend module).

## Proposal

- [x] make the logger an instance property of the backend class to allow a child backend to overwrite the logger of its parent
- [x] move `HistoryMixin` to the data backed module to allow its logger to be overwritten too.

**Note**: This PR depends on #510

### CODE REVIEW

File: src/ralph/backends/data/base.py [hunk 1]
```diff
@@ -79,6 +77,15 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class Loggable:
+    """A class with a `logger` property."""
+
+    @functools.cached_property
```
Comment (wilbrdt) [inline src/ralph/backends/data/base.py:83]: nit: We could import `cached_property` from `functools`
```suggestion
    @cached_property
```

**Review by wilbrdt — APPROVED** [review]
Alt text

File: src/ralph/backends/data/base.py (hunk 1 above)
Comment (SergioSim) [reply src/ralph/backends/data/base.py:83]: Well spotted) Thanks) I have updated [this](https://github.com/openfun/ralph/pull/521/files#diff-abc4154695f25bd1399562f7c5e9f27fe9fba44ce29e46d606b46f1871b932a6R89) in PR #521 - could we postpone the change here?

Comment (SergioSim) [reply src/ralph/backends/data/base.py:83]: After thought, I think it would be better to make this change here) will rebase #521 )

Comment (wilbrdt) [reply src/ralph/backends/data/base.py]: Alright, as you prefer! :raised_hands:

**Pull request merged**